### PR TITLE
docs: convert `ErrorHandling` from `.rst` to `.md`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Resolves partially #49997.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

File: docs/ErrorHandling.md [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+
+There is one minor usability problem here, though. If the closure
+contains throwing expressions, those expressions must be explicitly
+marked within the closure with `try`. However, from the compiler\'s
+perspective, the call to `autoreleasepool` is also a call that can
+throw, and so it must also be marked with `try`:
+
+```swift
+try autoreleasepool {    // 'try' is required here...
+    let string = try parseString() // ...and here.
```
Comment (AnthonyLatsis) [inline docs/ErrorHandling.md:698]: Ditto.

File: docs/ErrorHandling.md [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+is fully applied and none of the function arguments can throw. For
+example:
+
+    // This call to map is considered not to throw because its
+    // argument function does not throw.
+    let absolutePaths = paths.map { "/" + $0 }
+
+    // This call to map is considered to throw because its
+    // argument function does throw.
+    let streams = try absolutePaths.map { try InputStream(filename: $0) }
```
Comment (AnthonyLatsis) [inline docs/ErrorHandling.md:236]: Overlooked code block.

File: docs/ErrorHandling.md [hunk 3]
```diff
@@ -0,0 +1,745 @@
...
+-   `NSFileCoordinator` has half a dozen methods where the `error:`
+    clause is second-to-last, followed by a block argument. These
+    methods are not deprecated as far as I know.
+
+The above translation rule would import methods like this one from
+`NSDocument`:
+
+    ```objc
+    - (NSDocument *)duplicateAndReturnError:(NSError **)outError;
+    ```
```
Comment (AnthonyLatsis) [inline docs/ErrorHandling.md:606]: This and the next three code blocks are not rendered properly, probably due to the indentation.

**Review by AnthonyLatsis — APPROVED** [review]
Thanks! Why are all the apostrophes and quotes escaped?

**Comment by AnthonyLatsis** [comment]
I suspect the diff is all red because some functions in the snippets don’t have a closing brace.

**Comment by MaxDesiatov** [comment]
> Thanks! Why are all the apostrophes and quotes escaped?

That's the behaviour of the tool that I use for conversion. Unfortunately, I'm not sure if it's customizable, and I'd prefer to stay on the safe side in case something actually has to be escaped. Backslashes before apostrophes and quotes have no effect on how those are rendered, so it seems safe to keep them.

File: docs/ErrorHandling.md [hunk 4]
```diff
@@ -0,0 +1,745 @@
...
+
+As a tentpole feature for Swift 2.0, we are introducing a new
+first-class error handling model. This feature provides standardized
+syntax and language affordances for throwing, propagating, catching, and
+manipulating recoverable error conditions.
+
+Error handling is a well-trod path, with many different approaches in
+other languages, many of them problematic in various ways. We believe
+that our approach provides an elegant solution, drawing on the lessons
+we\'ve learned from other languages and fixing or avoiding some of the
```
Comment (CodaFi) [inline docs/ErrorHandling.md:11]: Do we need to escape `'`s?

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test and merge

**Pull request merged**